src/retrieval/search_cache.py

```
@@ -0,0 +1,133 @@
+        similarity_threshold: Optional[float],
+        config: SearchConfig
+    ) -> str:
+        """Generate cache key for query results."""
+        # Create a string representation of the search parameters
+        cache_data = f"{query}:{top_k}:{similarity_threshold or config.similarity_threshold}"
+        
+        if search_filter:
+            cache_data += f":{search_filter.content_types}:{search_filter.confidence_threshold}"
+        
+        # Generate hash
+        return hashlib.md5(cache_data.encode()).hexdigest()
```

Ruff (0.13.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)